@jskeet @BillWagner I see in the TG2 meeting minutes from 2022-11-02, "We decided to include in the appropriate extraction templates a set of three or four commonly used namespaces (such as System, System.Collections, System.Collections.Generic, and System.Io), and to not use fully qualified names unless they are needed to support discussions on the use of such names."

I just searched the md files for explicit `using` namespace directives, and found the following:

```csharp
using System;
using System.Collections.Generic;
using System.Collections;
using System.Diagnostics;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using System.Security.Permissions;
using System.Text;
using System.Threading;
```

I added this list to the top of (my private version of) the templates standalone-console, standalone-lib, and code-in-main. I then tested a few examples that needed one or more of these.

Now when these templates are used with an example that already explicitly has one or more of these directives, I get

> `warning CS0105: The using directive for '<namespace name>' appeared previously in this namespace`

If we use this approach, we can look at all the examples having explicit `using` directives, and if those are *not necessary* for explaining the example, we can remove them; otherwise, we leave them in and add `ignoredWarnings:["CS0105"]` to the metadata. And, of course, we remove *all* fully qualified prefixes from examples where they are not necessary for explaining the example.

How does that sound?

--- a/tools/example-templates/code-in-main/Program.cs
+++ b/tools/example-templates/code-in-main/Program.cs
@@ -1,3 +1,15 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Permissions;
+using System.Text;
+using System.Threading;
+
 class Program
 {
     static void Main()

--- a/tools/example-templates/standalone-console/Program.cs
+++ b/tools/example-templates/standalone-console/Program.cs
@@ -1,1 +1,13 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Permissions;
+using System.Text;
+using System.Threading;
+
 $example-code

--- a/tools/example-templates/standalone-lib/Library.cs
+++ b/tools/example-templates/standalone-lib/Library.cs
@@ -1,1 +1,13 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Permissions;
+using System.Text;
+using System.Threading;
+
 $example-code

